### ISSUE TITLE
GSV: Invalid satellite status for empty fields 

### ISSUE THREAD

**Issue by norton-dev** [mariokonrad/marnav#37]
Hi,

Could You please check use case for from UT below. It is failing.
Probably because of falsely positive satellite_info structure. 

```
TEST(MarnavNmeaTests, GsvGetSatEmptySats)
{
    auto sentence = marnav::nmea::make_sentence(
        "$GLGSV,3,3,10,83,11,003,,83,11,003,,,,,,,,,*64");
    auto gsvSentence = marnav::nmea::sentence_cast<marnav::nmea::gsv>(sentence);

    EXPECT_EQ(3u, gsvSentence->get_n_messages());
    EXPECT_EQ(3u, gsvSentence->get_message_number());
    EXPECT_EQ(10u, gsvSentence->get_n_satellites_in_view());

    {
        const auto sat = *gsvSentence->get_sat(0);
        EXPECT_EQ(83u, sat.prn);
        EXPECT_EQ(11u, sat.elevation);
        EXPECT_EQ(3u, sat.azimuth);
        EXPECT_FALSE(sat.snr);
    }
    {
        const auto sat = *gsvSentence->get_sat(1);
        EXPECT_EQ(83u, sat.prn);
        EXPECT_EQ(11u, sat.elevation);
        EXPECT_EQ(3u, sat.azimuth);
        EXPECT_FALSE(sat.snr);
    }
    {
        const auto sat = gsvSentence->get_sat(2);
        EXPECT_FALSE(sat.has_value());
    }
    {
        const auto sat = gsvSentence->get_sat(3);
        EXPECT_FALSE(sat.has_value());
    }
}
```

**Comment by mariokonrad** [comment]
Hello

Thanks for pointing this out. The error is in the initialization of the satellite info in gsv (constructor).
Needs to be addressed, and I'll fix it.

**Comment by mariokonrad** [comment]
Should be fixed in d7329a970001af697e25e3c0bc2ee4df2c954b66, although I'm not necessarily super happy with the solution.
For now it it appears to work.

**Comment by mariokonrad** [comment]
Closing, due to fixed and lack of feedback.
